pkg/ingestor/verifier/verifier.go

```diff
@@ -0,0 +1,42 @@
+//
+// Copyright 2022 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package verifier
+
+import (
+	"crypto"
+)
+
+// The purpose of Verifier is to allow for multiple signature or identity
+// verifiers that will check envelopes, signatures, and other payloads
+// for linking a payload back to an identity. In most cases this will be
+// done through verifying one or more signatures against known public keys.
+type Verifier interface {
+	// Verify takes in bytes and returns a list of identities. Those
+	// identities are marked as verified if the payload can be tied back to
+	// an identity, usually through signature validation. In cases where
+	// signatures can't be verified, it is still valuable to return data
+	// about the unverified identity.
+	Verify(payloadBytes []byte) ([]Identity, error)
+	Type() string
+}
+
+// Elements of this struct might be nil/empty if the key is invalid or the
+// Id of the identity can't be determined.
+type Identity struct {
+	Id       string
+	Key      crypto.PublicKey
```

Comment:
maybe add a comment here that this should only be used if `Verified` is true?

Reply:
What do you mean?